rclcpp/test/rclcpp/test_qos_event.cpp

```diff
@@ -112,34 +117,42 @@ TEST_F(TestQosEvent, test_publisher_constructor)
 /*
    Testing construction of a subscriptions with QoS event callback functions.
  */
-TEST_F(TestQosEvent, test_subscription_constructor)
+TEST_F(TestQosEvent, test_subscription_constructor_with_event_callbacks)
 {
+  // While rmw_zenoh does not support Deadline/LivelinessChanged events,
+  // it does support IncompatibleQoS
+  std::string rmw_implementation_str = std::string(rmw_get_implementation_identifier());
+
   rclcpp::SubscriptionOptions options;
 
   // options arg with no callbacks
   auto subscription = node->create_subscription<test_msgs::msg::Empty>(
     topic_name, 10, message_callback, options);
 
-  // options arg with one of the callbacks
-  options.event_callbacks.deadline_callback =
-    [node = node.get()](rclcpp::QOSDeadlineRequestedInfo & event) {
-      RCLCPP_INFO(
-        node->get_logger(),
-        "Requested deadline missed - total %d (delta %d)",
-        event.total_count, event.total_count_change);
-    };
+  if (rmw_implementation_str != "rmw_zenoh_cpp") {
+    // options arg with one of the callbacks
+    options.event_callbacks.deadline_callback =
+      [node = node.get()](rclcpp::QOSDeadlineRequestedInfo & event) {
+        RCLCPP_INFO(
+          node->get_logger(),
+          "Requested deadline missed - total %d (delta %d)",
+          event.total_count, event.total_count_change);
+      };
+  }
   subscription = node->create_subscription<test_msgs::msg::Empty>(
     topic_name, 10, message_callback, options);
 
-  // options arg with two of the callbacks
-  options.event_callbacks.liveliness_callback =
-    [node = node.get()](rclcpp::QOSLivelinessChangedInfo & event) {
-      RCLCPP_INFO(
-        node->get_logger(),
-        "Liveliness changed - alive %d (delta %d), not alive %d (delta %d)",
-        event.alive_count, event.alive_count_change,
-        event.not_alive_count, event.not_alive_count_change);
-    };
+  if (rmw_implementation_str != "rmw_zenoh_cpp") {
```

Comment:
Combine within `if` block above.

Reply:
we can't combine them, because the subscription constructor use options with different setups.